# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: accessible/ipc/RemoteAccessible.cpp
# Commit: 4570141f12e9
# Full Hash: 4570141f12e9018910dcff1b57fc8070a8601ede
# Author: Morgan Rae Reschenberg <mreschenberg@mozilla.com>
# Date: 2023-06-29 21:48:38
# Description:
#   Bug 1840343: Track containers and text leaves separately when fuzzy hittesting r=Jamie
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D182269
# ==============================================================================

diff -r 7569a78f5815 -r 4570141f12e9 accessible/ipc/RemoteAccessible.cpp
--- a/accessible/ipc/RemoteAccessible.cpp	Thu Jun 29 18:01:25 2023 +0000
+++ b/accessible/ipc/RemoteAccessible.cpp	Thu Jun 29 18:09:46 2023 +0000
@@ -453,26 +453,27 @@
   // If we found a clipped container, descend it in search of
   // meaningful text leaves. Ignore non-text-leaf/text-container
   // siblings.
-  RemoteAccessible* maybeTextLeaf = clippedContainer;
-  while (maybeTextLeaf) {
+  RemoteAccessible* container = clippedContainer;
+  while (container) {
+    RemoteAccessible* textLeaf = nullptr;
     bool continueSearch = false;
-    childCount = maybeTextLeaf->ChildCount();
+    childCount = container->ChildCount();
     for (uint32_t i = 0; i < childCount; i++) {
-      RemoteAccessible* child = maybeTextLeaf->RemoteChildAt(i);
+      RemoteAccessible* child = container->RemoteChildAt(i);
       if (child->Role() == roles::TEXT_CONTAINER) {
-        maybeTextLeaf = child;
+        container = child;
         continueSearch = true;
         break;
       }
       if (child->IsTextLeaf()) {
-        maybeTextLeaf = child;
+        textLeaf = child;
         // Don't break here -- it's possible a text container
         // exists as another sibling, and we should descend as
         // deep as possible.
       }
     }
-    if (maybeTextLeaf && maybeTextLeaf->IsTextLeaf()) {
-      return maybeTextLeaf;
+    if (textLeaf) {
+      return textLeaf;
     }
     if (!continueSearch) {
       // We didn't find anything useful in this set of siblings.